flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/sink/committer/FilesCommitter.java

```diff
@@ -0,0 +1,278 @@
+    if (tableLoader != null) {
+      tableLoader.close();
+    }
+
+    if (workerPool != null) {
+      workerPool.shutdown();
+    }
+  }
+
+  private void commitResult(
+      int dataFilesNum,
+      int deleteFilesNum,
+      Collection<CommitRequest<FilesCommittable>> committableCollection) {
+    int totalFiles = dataFilesNum + deleteFilesNum;
+
+    if (totalFiles != 0) {
+      if (replacePartitions) {
+        replacePartitions(committableCollection, dataFilesNum, deleteFilesNum);
+      } else {
+        commitDeltaTxn(committableCollection, dataFilesNum, deleteFilesNum);
+      }
+    }
+  }
+
+  private void replacePartitions(
+      Collection<CommitRequest<FilesCommittable>> writeResults,
+      int dataFilesNum,
+      int deleteFilesNum) {
+    // Partition overwrite does not support delete files.
+    Preconditions.checkState(deleteFilesNum == 0, "Cannot overwrite partitions with delete files.");
+
+    // Commit the overwrite transaction.
+    ReplacePartitions dynamicOverwrite = table.newReplacePartitions().scanManifestsWith(workerPool);
+
+    String flinkJobId = null;
+    long checkpointId = -1;
+    for (CommitRequest<FilesCommittable> commitRequest : writeResults) {
+      WriteResult result = commitRequest.getCommittable().committable();
+      Preconditions.checkState(
+          result.referencedDataFiles().length == 0, "Should have no referenced data files.");
+      flinkJobId = commitRequest.getCommittable().jobID();
+      checkpointId = commitRequest.getCommittable().checkpointId();
+      Arrays.stream(result.dataFiles()).forEach(dynamicOverwrite::addFile);
+    }
+
+    commitOperation(
+        writeResults,
+        dynamicOverwrite,
+        dataFilesNum,
+        0,
+        "dynamic partition overwrite",
+        flinkJobId,
+        checkpointId);
+  }
+
+  private void commitDeltaTxn(
+      Collection<CommitRequest<FilesCommittable>> writeResults,
+      int dataFilesNum,
+      int deleteFilesNum) {
+    if (deleteFilesNum == 0) {
+      // To be compatible with iceberg format V1.
+      AppendFiles appendFiles = table.newAppend().scanManifestsWith(workerPool);
+      String flinkJobId = null;
+      long checkpointId = -1;
+      for (CommitRequest<FilesCommittable> commitRequest : writeResults) {
+        WriteResult result = commitRequest.getCommittable().committable();
+        Preconditions.checkState(
+            result.referencedDataFiles().length == 0, "Should have no referenced data files.");
+        Arrays.stream(result.dataFiles()).forEach(appendFiles::appendFile);
+        flinkJobId = commitRequest.getCommittable().jobID();
+        checkpointId = commitRequest.getCommittable().checkpointId();
+      }
+
+      commitOperation(
+          writeResults, appendFiles, dataFilesNum, 0, "append", flinkJobId, checkpointId);
+
+    } else {
+      // To be compatible with iceberg format V2.
+      for (CommitRequest<FilesCommittable> commitRequest : writeResults) {
```

Comment:
The order of the `CommitRequests` are important, so the `commitRequest` object should be a `List`

Reply:
It's a `List` instance, just not declared as a `List`. I've changed its type.